src/Tribe/REST/V1/Headers/Base.php

```diff
@@ -81,7 +81,36 @@ public function get_api_origin_meta_name() {
 	 */
 	public function get_rest_url() {
 		if ( is_single() && tribe_is_event() ) {
-			return tribe_events_rest_url( 'events/' . Tribe__Main::post_id_helper() );
+			$this_post_id = Tribe__Main::post_id_helper();
+
+			/**
+			 * We are dealing with an /all kind of query.
+			 * The root URL has to be modified to include the
+			 * post parent and its children post IDs.
+			 */
+			if (
+				( $wp_query = tribe_get_global_query_object() )
+				&& ( $post_parent = $wp_query->get( 'post_parent' ) )
+				&& $post_parent == $this_post_id
+			) {
+				$all      = array( $post_parent );
+				$children = get_posts( array(
+					'post_type'      => Tribe__Events__Main::POSTTYPE,
+					'fields'         => 'ids',
+					'posts_per_page' => - 1,
```

Comment:
Maybe might be a good idea to set a limit here and paginate on this endpoint instead to avoid having a slow endpoint with ~ 1k events

Reply:
This is a very good idea.